pkg/server/server.go

```diff
@@ -31,6 +33,20 @@ func New() *Server {
 }
 
 func (s *Server) ListenAndServe(addr string) error {
+	// Server should remove the socket file prior to binding it in case the socket isn't cleaned up gracefully.
+	// This can happen if the application is killed by SIGKILL or SIGSTOP, i.e. kill -9 or docker kill by default.
+	_, err := os.Stat(addr)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("failed to os.Stat socket: %v", err)
+	}
+	if err == nil {
+		// the socket file exists, it should be removed
+		zap.L().Info("Removing existing socket", zap.String("address", addr))
+		err = os.Remove(addr)
+		if err != nil && !os.IsNotExist(err) {
+			return fmt.Errorf("failed to os.Remove existing socket: %v", err)
+		}
+	}
```

Comment:
@nnmin-aws see if restructuring the code a bit makes it more read-able?

```suggestion
	// Server should remove the socket file prior to binding it in case the socket isn't cleaned up gracefully.
	// This can happen if the application is killed by SIGKILL or SIGSTOP, i.e. kill -9 or docker kill by default.
	if _, err := os.Stat(addr); err != nil {
		if !os.IsNotExist(err) {
			return fmt.Errorf("failed to os.Stat socket: %v", err)
		}
	} else {
		// the socket file exists, it should be removed
		zap.L().Info("Removing existing socket", zap.String("address", addr))
		if err = os.Remove(addr); err != nil {
			if !os.IsNotExist(err) {
				return fmt.Errorf("failed to os.Remove existing socket: %v", err)
			}
		}
	}
```

Reply:
thanks @dims for the comments! it looks great. I have updated